### PR TITLE
feat(docs): Create gatsby internationalization guide

### PR DESCRIPTION
**Pull request by audoralc**
Draft for issue #8716: Content migration - fill out "Gatsby i18n" stub article

Closes https://github.com/gatsbyjs/gatsby/issues/3853

### CODE REVIEW

File: docs/docs/gatsby-i18n.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+title: Gatsby i18n
+---
+Serving users content in a way that is adapted to their language & culture is part of a great user experience. When you make an effort to adapt web content to a users location that practice is called internationalization (i18n).
+
+In practice i18n involves translating text and formatting dates, numbers, and strings based on the user's detected location. For example, a date displayed for a user in the United States would follow the mm/dd/year date format, but for a user in the UK the date format would change to dd/mm/year.
+
+This guide is a brief look at the options that exist for enhancing your Gatsby project for internationalization.
+
+### Choosing a package
```
Comment (amberleyromo) [inline docs/docs/gatsby-i18n.md]: Can you generalize some of this away from your specific project use case? "We opted for", "we wanted". Would make sense in a blog post, but for a doc it makes more sense to be generalized.

**Comment by calcsam** [comment]
Hey @audoralc -- would love to get this in if you can make the changes @amberleyromo suggested!

**Comment by shannonbux** [comment]
How's this PR going? Would be cool to see this get merged!

File: docs/docs/gatsby-i18n.md (hunk 1 above)
Comment (audoralc) [reply docs/docs/gatsby-i18n.md]: Changed it to be much more general - if it's a little too much please let me know.

**Comment by LekoArts** [comment]
Hey @marcysutton 	👋 	
Would be great to have this in. Can you please have a look? Thanks!

File: docs/docs/gatsby-i18n.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+[gatsby-plugin-i18n on Github](https://github.com/angeloocana/gatsby-plugin-i18n)
+
+#### react-intl
+
+React-intl is a part of the FormatJS set of i18n libraries and provides support for over 150+ languages. It builds on JavaScript's [Internationalization API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) providing enhanced APIs and components. React-intl uses React context and HOCs (Higher Order Components) to provide translations allowing you to dynamically load language modules as you need them. There are also polyfill options available for older browsers that do not support the base JavaScript i18n API.
+
+More detailed information about react-intl's [APIs](https://github.com/yahoo/react-intl/wiki/API) and [components](https://github.com/yahoo/react-intl/wiki/Components), including [demos](https://github.com/yahoo/react-intl/tree/master/examples), are available in the [documentation](https://github.com/yahoo/react-intl/wiki).
+
+#### react-i18next
```
Comment (marcysutton) [inline docs/docs/gatsby-i18n.md]:
```suggestion
### react-i18next
```

File: docs/docs/gatsby-i18n.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+React-i18next is an internationalization library built on the i18next framework. It uses components to make sure translations render correctly or to re-render your content when the user language changes.
+
+React-i18next is more extensible than other options with a variety of plugins, utilities, and configurations. Common plugins allow for detecting a user's language or adding an additional layer of local caching. Other options include caching, a backend plugin to load translations from your server or bundling translations with Webpack.
```
Comment (marcysutton) [inline docs/docs/gatsby-i18n.md]:
```suggestion
React-i18next is more extensible than other options with a variety of plugins, utilities, and configurations. Common plugins allow for detecting a user's language or adding an additional layer of local caching. Other options include caching, a backend plugin to load translations from your server, or bundling translations with Webpack.
```

File: docs/docs/gatsby-i18n.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+This framework also has experimental support for the React suspense API and React hooks.
+
```
Comment (marcysutton) [inline docs/docs/gatsby-i18n.md]: This doc would benefit from a conclusion to wrap up. It does a good job of setting up choosing a package earlier on, but as a reader I still don't know which one I would choose by the end of the doc. Is there any more information we can provide developers to help them select an i18n package?

File: docs/docs/gatsby-i18n.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
+---
+title: Gatsby i18n
```
[the remark this answers is not in the document]
Comment (jlengstorf) [reply docs/docs/gatsby-i18n.md]: Maybe even include both? `Gatsby Internationalization (i18n)`

Could be good for SEO/site searches.

[the remark this answers is not in the document]
Comment (marcysutton) [reply docs/docs/gatsby-i18n.md]: That was my first suggestion, and then I saw it was mentioned in the intro paragraph and simplified it further. That would totally work though!

**Comment by marcysutton** [comment]
Ok, I think this is ready to go! I didn't end up adding a conclusion to it, as I'm not familiar enough with these packages to write it. But we could always add something on later! Thanks again for your work on this PR, @audoralc! 🙌

File: www/src/data/sidebars/doc-links.yaml [hunk 6]
```diff
@@ -305,8 +305,8 @@
               link: /docs/seo/
             - title: Optimizing Site Performance with Guess.js
               link: /docs/optimizing-site-performance-with-guessjs/
-        - title: Localizing Your Site*
-          link: /docs/localization-i18n/
+        - title: Internationalizing Your Site (i18n)
```
Comment (marcysutton) [inline www/src/data/sidebars/doc-links.yaml]: Any feedback on this title change? The distinction between localization and internationalization is sorta unclear to me even after reading up on it. My hunch is that "internationalizing" doesn't make sense as a verb, but I was hesitant to make a switch to "localizing" in the doc. Happy to revisit this next week, if not sooner. https://www.w3.org/International/questions/qa-i18n

File: www/src/data/sidebars/doc-links.yaml [hunk 7]
```diff
@@ -305,8 +305,8 @@
               link: /docs/seo/
             - title: Optimizing Site Performance with Guess.js
               link: /docs/optimizing-site-performance-with-guessjs/
-        - title: Localizing Your Site*
-          link: /docs/localization-i18n/
+        - title: Localization and Internationalization with Gatsby
+          link: /docs/gatsby-i18n/
```
Comment (wardpeet) [inline www/src/data/sidebars/doc-links.yaml]: When changing the link it's best to also add a redirect to 
https://github.com/gatsbyjs/gatsby/blob/master/www/gatsby-node.js#L228

If you have no idea how to do that, just give a shout and I can do it for you. ❤️ 

(Great job on the rest of the PR!)

Comment (marcysutton) [reply www/src/data/sidebars/doc-links.yaml]: It was a stub, so it didn't really seem necessary to me!

Comment (marcysutton) [reply www/src/data/sidebars/doc-links.yaml]: I ended up changing the link back to the original (just changed the title), so no redirect needed!

**Comment by marcysutton** [comment]
Thanks so much for your contribution, @audoralc! This is a great addition to the Gatsby docs!

**Pull request merged**